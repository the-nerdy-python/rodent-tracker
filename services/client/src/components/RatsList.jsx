import React from 'react';

const RatsList = (props) => {
  return (
    <div>
      {
        props.rats.map((rat) => {
          return (
            <h4
              key={rat.id}
              className="box title is-4"
            >{ rat.color }
            </h4>
          )
        })
      }
    </div>
  )
};

export default RatsList;
