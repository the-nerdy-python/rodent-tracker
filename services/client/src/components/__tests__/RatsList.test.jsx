import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import RatsList from '../RatsList';

const rats = [
  {
    'active': true,
    'color': 'white',
    'id': 1,
    'weight': 200
  },
  {
    'active': true,
    'color': 'black',
    'id': 2,
    'weight': 300
  }
];

test('RatsList renders properly', () => {
  const wrapper = shallow(<RatsList rats={rats}/>);
  const element = wrapper.find('h4');
  expect(element.length).toBe(2);
  expect(element.get(0).props.children).toBe('white');
});

test('RatsList renders a snapshot properly', () => {
    const tree = renderer.create(<RatsList rats={rats}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

