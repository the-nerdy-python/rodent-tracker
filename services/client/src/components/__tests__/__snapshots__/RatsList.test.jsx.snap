// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatsList renders a snapshot properly 1`] = `
<div>
  <h4
    className="box title is-4"
  >
    white
  </h4>
  <h4
    className="box title is-4"
  >
    black
  </h4>
</div>
`;
